Mathlib/AlgebraicGeometry/Morphisms/ClosedImmersion.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright (c) 2023 Jonas van der Schaaf. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston, Christian Merten, Jonas van der Schaaf
+-/
+import Mathlib.AlgebraicGeometry.OpenImmersion
+import Mathlib.CategoryTheory.MorphismProperty.Composition
+import Mathlib.RingTheory.LocalProperties
+
+/-!
+
+# Closed immersions of schemes
+
+A morphism of schemes `f : X ⟶ Y` is a closed immersion if the underlying map of topological spaces
+is a closed immersion and the induced morphisms of stalks are all surjective.
+
+## Main definitions
+
+* `IsClosedImmersion` : The property of scheme morphisms stating `f : X ⟶ Y` is a closed immersion.
+
+## TODO
+
+* Show closed immersions of affines are induced by surjective ring maps
+* Show closed immersions are stable under pullback
+* Define closed immersions of locally ringed spaces, where we also assume that the kernel of `O_X →
+  f_*O_Y` is locally generated by sections as an `O_X`-module, and relate it to this file. See
+  https://stacks.math.columbia.edu/tag/01HJ.
```

Comment:
Other TODOs: one of my favourite theorems about closed immersions is that they're precisely the proper monomorphisms. One could imagine doing the easy direction :-)